### PR TITLE
ACA-133: Update error code mapping for NoC APIs to reflect LLD changes

### PR DESCRIPTION
**Pull request by Abi-Laver**
https://tools.hmcts.net/jira/browse/ACA-133

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/functionalTest/resources/features/F-206 - Request Notice Of Change/S-206.21.td.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "_extends_": "ACA_400_Response",
     "body": {
       "status" : "BAD_REQUEST",
-      "message": "Insufficient privileges for notice of change request",
+      "message" : "The number of provided answers must match the number of questions - expected 2 answers, received 1",
```
Comment (RebeccaBaker) [inline src/functionalTest/resources/features/F-206 - Request Notice Of Change/S-206.21.td.json:39]: why has the message changed?

File: src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/noc/NoCException.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.reform.managecase.api.errorhandling.noc;
+
+@SuppressWarnings({"PMD.MissingSerialVersionUID"})
+public class NoCException extends RuntimeException {
+    public String errorCode;
+    public String errorMessage;
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/noc/NoCException.java:6]: would be good to resolve the code smells on this class

**Review by RebeccaBaker — COMMENTED** [review]
some minor comments

File: src/functionalTest/resources/features/F-206 - Request Notice Of Change/S-206.21.td.json (hunk 1 above)
Comment (Abi-Laver) [reply src/functionalTest/resources/features/F-206 - Request Notice Of Change/S-206.21.td.json:39]: A different error is being thrown now, also occurring in the NoC branch

File: src/main/java/uk/gov/hmcts/reform/managecase/api/controller/NoticeOfChangeController.java [hunk 3]
```diff
@@ -425,13 +427,14 @@ private void validateCaseIds(String caseId) {
                 + "\n2) " + CASE_ID_INVALID_LENGTH
                 + "\n3) " + CASE_ID_EMPTY
                 + "\n4) " + CHALLENGE_QUESTION_ANSWERS_EMPTY
-                + "\n5) " + MISSING_COR_CASE_ROLE_ID_IN_CASE_DEFINITION,
-            response = ApiError.class,
+                + "\n5) " + "Missing ChangeOrganisationRequest.CaseRoleID %s in the case definition",
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/managecase/api/controller/NoticeOfChangeController.java:430]:
```suggestion
                + "\n5) " + "Missing ChangeOrganisationRequest.CaseRoleID [APPLICANT] in the case definition",
```

**Review by danlysiak — COMMENTED** [review]
A few little comments

**Review by danlysiak — APPROVED** [review]

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**